samples/nrf9160/modem_shell/src/link/link_shell.c

```diff
@@ -640,251 +571,83 @@ int link_shell_get_and_print_current_system_modes(
-	} else if (strcmp(argv[1], "coneval") == 0) {
-		link_cmd_args.command = LINK_CMD_CONEVAL;
-	} else if (strcmp(argv[1], "rsrp") == 0) {
-		require_subscribe = true;
-		link_cmd_args.command = LINK_CMD_RSRP;
-	} else if (strcmp(argv[1], "search") == 0) {
-		require_option = true;
-		link_cmd_args.command = LINK_CMD_SEARCH;
-	} else if (strcmp(argv[1], "ncellmeas") == 0) {
-		require_option = true;
-		link_cmd_args.command = LINK_CMD_NCELLMEAS;
-	} else if (strcmp(argv[1], "msleep") == 0) {
-		require_option = true;
-		link_cmd_args.command = LINK_CMD_MDMSLEEP;
-	} else if (strcmp(argv[1], "tau") == 0) {
-		require_subscribe = true;
-		link_cmd_args.command = LINK_CMD_TAU;
-	} else if (strcmp(argv[1], "connect") == 0) {
-		require_apn = true;
-		link_cmd_args.command = LINK_CMD_CONNECT;
-	} else if (strcmp(argv[1], "disconnect") == 0) {
-		require_pdn_cid = true;
-		link_cmd_args.command = LINK_CMD_DISCONNECT;
-	} else if (strcmp(argv[1], "defcont") == 0) {
-		link_cmd_args.command = LINK_CMD_DEFCONT;
-	} else if (strcmp(argv[1], "defcontauth") == 0) {
-		link_cmd_args.command = LINK_CMD_DEFCONTAUTH;
-	} else if (strcmp(argv[1], "funmode") == 0) {
-		require_option = true;
-		link_cmd_args.command = LINK_CMD_FUNMODE;
-	} else if (strcmp(argv[1], "sysmode") == 0) {
-		require_option = true;
-		link_cmd_args.command = LINK_CMD_SYSMODE;
-	} else if (strcmp(argv[1], "nmodeat") == 0) {
-		link_cmd_args.command = LINK_CMD_NORMAL_MODE_AT;
-	} else if (strcmp(argv[1], "nmodeauto") == 0) {
-		link_cmd_args.command = LINK_CMD_NORMAL_MODE_AUTO;
-	} else if (strcmp(argv[1], "edrx") == 0) {
-		require_option = true;
-		link_cmd_args.command = LINK_CMD_EDRX;
-	} else if (strcmp(argv[1], "psm") == 0) {
-		require_option = true;
-		link_cmd_args.command = LINK_CMD_PSM;
-	} else if (strcmp(argv[1], "rai") == 0) {
-		require_option = true;
-		link_cmd_args.command = LINK_CMD_RAI;
-	} else if (strcmp(argv[1], "dnsaddr") == 0) {
-		link_cmd_args.command = LINK_CMD_DNSADDR;
-	} else if (strcmp(argv[1], "redmob") == 0) {
-		link_cmd_args.command = LINK_CMD_REDMOB;
-	} else {
-		mosh_error("Unsupported command=%s\n", argv[1]);
-		ret = -EINVAL;
-		goto show_usage;
-	}
-
-	/* Reset getopt due to possible previous failures: */
-	optreset = 1;
-
-	/* We start from subcmd arguments */
-	optind = 2;
-
 	int long_index = 0;
 	int opt;
-	int apn_len = 0;
-
-	char edrx_value_str[LINK_SHELL_EDRX_VALUE_STR_LENGTH + 1];
-	bool edrx_value_set = false;
-	char edrx_ptw_bit_str[LINK_SHELL_EDRX_PTW_STR_LENGTH + 1];
-	bool edrx_ptw_set = false;
 
-	char psm_rptau_bit_str[LINK_SHELL_PSM_PARAM_STR_LENGTH + 1];
-	bool psm_rptau_set = false;
-	char psm_rat_bit_str[LINK_SHELL_PSM_PARAM_STR_LENGTH + 1];
-	bool psm_rat_set = false;
-
-	char *normal_mode_at_str = NULL;
-	uint8_t normal_mode_at_mem_slot = 0;
-
-	while ((opt = getopt_long(argc, argv,
-				  "a:I:f:i:x:w:p:t:A:P:U:su014rmngMNed",
-				  long_options, &long_index)) != -1) {
+	optreset = 1;
```

Comment:
Looking at getopt_long specification, it doesn't take advantage of optreset because you cannot call it for same arguments twice anyway due to permutations. Basic getopt() would take advantage of it but we are not using it. So it's not necessary here nor in any getopt_long calls we have in mosh.